nx/lib/nx/template_backend.ex

```diff
@@ -11,8 +11,12 @@ defmodule Nx.TemplateBackend do
   defstruct []
 
   @impl true
-  def inspect(%Nx.Tensor{}, _opts) do
-    "Nx.TemplateBackend"
+  def inspect(%Nx.Tensor{}, opts) do
+    if opts.custom_options[:add_backend_on_inspect] do
+      "Nx.TemplateBackend"
+    else
+      ""
```

Comment:
```suggestion
      Inspect.Algebra.empty()
```

Reply:
Actually, this logic should be on Nx.Tensor, so it applies to all backends.

Reply:
I'll try something out :)

Reply:
And this one should still apply!